File: StrategyGeneration/src/main/java/com/finance/strategyGeneration/model/StatisticsOfStrategy.java
package com.finance.strategyGeneration.model;

import lombok.*;
import lombok.experimental.FieldDefaults;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

@Getter
@Builder
@RequiredArgsConstructor
@Table("statistics_of_strategy")
@EqualsAndHashCode
@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
public class StatisticsOfStrategy {

    @Id
    long id;
    long specificationOfStrategyId;
    long score;
    int valueOfAcceptableRisk;
    long maximumPercentDrawdownFromStartScore;
    long averagePercentDrawdownOfScore;
    long maximumValueFromScore;
    int numberOfWinningDeals;
    int numberOfLosingDeals;

}

File: StatisticsService/src/main/kotlin/com/finance/strategy/statistics/model/StatisticsOfStrategy.kt
package com.finance.strategy.statistics.model

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table

@Table("statistics_of_strategy")
data class StatisticsOfStrategy(
    @Id
    val id: Long = 0,
    val specificationOfStrategyId: Long = 0,
    val score: Long = 0,
    val valueOfAcceptableRisk: Int = 0,
    val maximumPercentDrawdownFromStartScore: Long = 0,
    val averagePercentDrawdownOfScore: Long = 0,
    val maximumValueFromScore: Long = 0,
    val numberOfWinningDeals: Int = 0,
    val numberOfLosingDeals: Int = 0
)
